File: Core_PetBoarding_Backend/PetBoarding_Api/Dto/Prestations/PrestationDto.cs
namespace PetBoarding_Api.Dto.Prestations;

using PetBoarding_Domain.Prestations;

public record PrestationDto
{
    public Guid Id { get; init; }
    public string Libelle { get; init; } = string.Empty;
    public string Description { get; init; } = string.Empty;
    public TypeAnimal CategorieAnimal { get; init; }
    public decimal Prix { get; init; }
    public int DureeEnMinutes { get; init; }
    public bool EstDisponible { get; init; }
    public DateTime DateCreation { get; init; }
    public DateTime? DateModification { get; init; }
}

File: ng_PetBoarding_app/src/app/shared/contracts/prestations/prestation.dto.ts
import { PetType } from '../../../features/pets/models/pet.model';

export interface PrestationDto {
  id: string;
  libelle: string;
  description: string;
  categorieAnimal: PetType;
  prix: number;
  dureeEnMinutes: number;
  estDisponible: boolean;
  dateCreation: string;
  dateModification?: string;
}

export interface CreatePrestationRequest {
  libelle: string;
  description: string;
  categorieAnimal: PetType;
  prix: number;
  dureeEnMinutes: number;
  estDisponible: boolean;
}

export interface UpdatePrestationRequest {
  libelle?: string;
  description?: string;
  categorieAnimal?: PetType;
  prix?: number;
  dureeEnMinutes?: number;
  estDisponible?: boolean;
}

export interface GetAllPrestationsResponse {
  prestations: PrestationDto[];
  totalCount: number;
}

export interface GetPrestationResponse {
  prestation: PrestationDto;
}

export interface CreatePrestationResponse {
  prestation: PrestationDto;
}

export interface UpdatePrestationResponse {
  prestation: PrestationDto;
}

export interface DeletePrestationResponse {
  success: boolean;
  message: string;
}
